### PR TITLE
Feature/clean architecture

### PR DESCRIPTION
**Pull request by JanFidor**
## What type of PR is this? (check all applicable)

- [ ] 🚀 Added Name
- [x] ✨ Feature
- [ ] 🌟 stared the repo
- [ ] 🐛 Grammatical Error
- [ ] 📝 Documentation Update
- [x] 🚩 Other

## Description
Added a new module with clean architecture principals (including MVVM, DI, fragments, nav component, kotlinx serialization and more!)

## Add Link of GitHub Profile
https://github.com/JanFidor

### CODE REVIEW

**Comment by Tc2r1** [comment]
@JanFidor  You are my favorite person to have committed so far, Thanks for taking the time to add a module! AWESOME JOB! I'll review it later today but I'm sure many developers will be able to learn a lot from this. 

Btw, did you get a chance to implement a menu fragment/view so users could select other topics besides the languages/android quiz? 

Please add screen shots of the UI to this PR, Great work again!.

File: Interview_Prep_App/native_kotlin/clean-architecture/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    kotlin("plugin.serialization")
+    id("kotlin-parcelize")
+    id("androidx.navigation.safeargs.kotlin")
+}
+
+android {
+    compileSdk = rootProject.extra.get("compileSdkVersion") as Int
+
+    defaultConfig {
+        applicationId = "com.interviewprep.template"
```
Comment (Tc2r1) [inline Interview_Prep_App/native_kotlin/clean-architecture/build.gradle.kts:15]: Please rename to something you like better, or "com.interviewPrep.kotlin_clean" lol

File: Interview_Prep_App/native_kotlin/clean-architecture/src/main/assets/answers.json [hunk 2]
```diff
@@ -0,0 +1,144 @@
+{
```
Comment (Tc2r1) [inline Interview_Prep_App/native_kotlin/clean-architecture/src/main/assets/answers.json:1]: why did you put the .jsons here? Are you still getting them from the github repo later?

**Review by Tc2r1 — CHANGES_REQUESTED** [review]
Screen Shot 2022-10-24 at 9 45 44 AM

When you reach the last question, there is no way to answer it or see the score board? 

error
One of the Questions has the details as the question text? 

Submit Answer button needs bottom margin from parent. 
Screen Shot 2022-10-24 at 9 49 56 AM

**Comment by Tc2r1** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/tc2r1/hacktoberfest_interview_app) found an issue: Parentheses in (correctAnswersCount / questionsCount) are unnecessary and can be replaced with: correctAnswersCount / questionsCount

It's currently on:
[Interview_Prep_App\native_kotlin\clean-architecture\src\main\java\com\interviewprep\clean_architecture\ui\score\FinalScore.kt:11](https://github.com/Tc2r1/Hacktoberfest_Interview_App/blob/327c49f06644ed95e1f6e368d0ae10c799cf3893/Interview_Prep_App/native_kotlin/clean-architecture/src/main/java/com/interviewprep/clean_architecture/ui/score/FinalScore.kt#L11)
Commit 327c49f06644ed95e1f6e368d0ae10c799cf3893

File: Interview_Prep_App/native_kotlin/clean-architecture/build.gradle.kts (hunk 1 above)
Comment (JanFidor) [reply Interview_Prep_App/native_kotlin/clean-architecture/build.gradle.kts:15]: Good catch, forgot about that hahaha

**Comment by JanFidor** [comment]
Hi @Tc2r1! I absolutely love the feedback and I have to say that writing the code was an absolute joy.  Your app had just the right amount of back-end wizardry, enough to decently show off clean architecture while being small enough to be easily understood.  And of course the UI was already written ;P

**Comment by Tc2r1** [comment]
@JanFidor HI JAN, I created a PR on your repo I think to help you out some, did you see it? can you join me for a short conversation :D https://8x8.vc/tc2r1

**Comment by JanFidor** [comment]
I'm a little swamped with work, studies and everything, but I might be able to find a few hours during the long weekend to code-up choice of topics. Might even do it in compose, because having relatively small nested composables is slowly starting to grow on me after some of my XML shenanigans with gigantic files hahaha. Navigation is still pretty bad there, but it's nothing a tactical fragment with nav component won't fix ;P

**Comment by Tc2r1** [comment]
@JanFidor 

Your code isn't working as intended, I'd like to show you why in a short call if possible but I've made some changes to help you get to that direction. 

https://github.com/JanFidor/Hacktoberfest_Interview_App/pull/1

Is there a reason you didn't create a backend to get the json from the github repo? Your code does not filter out the true or false questions and thus the quiz gets messed up when one is selected, can you please update with a fix so i can merge it, I want this to count for your hacktober!

**Comment by JanFidor** [comment]
> 

You might need to scroll down for submit button if the questions are too large
image

image

I also filtered out questions with empty shortAnswer

**Comment by JanFidor** [comment]
@Tc2r1  hi, I quickly added the trueOrFalse question handling and the button margin, here's a quick pic:
image

**Review by Tc2r1 — APPROVED** [review]
This Looks Great, Great Job!!!

**Pull request merged**